cpp/search/search.cpp

```diff
@@ -2003,7 +2004,11 @@ void Search::initNodeNNOutput(
   }
   if(isRoot && searchParams.rootNumSymmetriesToSample > 1) {
     vector<shared_ptr<NNOutput>> ptrs;
+    std::array<int, NNInputs::NUM_SYMMETRY_COMBINATIONS> symmetryIndexes;
+    std::iota(symmetryIndexes.begin(), symmetryIndexes.end(), 0);
     for(int i = 0; i<searchParams.rootNumSymmetriesToSample; i++) {
```

Comment:
Before, rootNumSymmetriesToSample > 8 would have been valid and meaningful. Now, it will give undefined behavior. This function should probably explictly check for that and fail, either via throwing an exception (KataGo code just uses a custom "StringError" exception for pretty much everything) or via an assert.

And additionally, the bounds code in setup.cpp should be tightened to reject values larger than 8 now.

Is it worth having a parameter to toggle with and without replacementness, to preserve old behavior? I'm not sure. I guess probably nobody was using values larger than 8 here in practice.

Reply:
I guess I made the changes in the other later PR instead of this one, but per @sanderland feedback I did change the config file processing to limit the maximum to NNInputs::NUM_SYMMETRY_COMBINATIONS. So it wouldn't be possible to specify one that was too high at startup. (https://github.com/roy7/KataGo/pull/1/files)

I doubt the old behavior is worth keeping around. Anyone using a # more than 1 currently, I'd assume without replacement is strictly better. And I just can't imagine someone using a number over 8 and intentionally wanting replacement instead of without.

Reply:
setup.cpp updated

Reply:
With setup updated to cap at NUM_SYMMETRY_COMBINATIONS I assume that should guarantee good behavior, but I can add an assert if you like.